src/SearchableDropdown/SearchableDropdown.tsx

```diff
@@ -0,0 +1,327 @@
+    width: '100%',
+  }),
+  listItem: css({
+    ':hover': {
+      backgroundColor: alpha(ColorPalette.background.bright, 0.5, true),
+    },
+  }),
+  listItems: (menuHeight: number) =>
+    css({
+      borderTop: `1px solid ${ColorPalette.lightGrey}`,
+      maxHeight: menuHeight - INPUT_FIELD_HEIGHT,
+      overflowY: 'auto',
+    }),
+  listWrapper: (menuHeight: number) =>
+    css({
+      boxShadow: '1px 1px 3px rgba(29, 29, 29, 0.125)',
+      maxHeight: menuHeight,
+      overflow: 'hidden',
+      width: '100%',
+      zIndex: 9999,
+    }),
+  wrapper: (isOpen: boolean, disabled: boolean, isLoading: boolean) =>
+    css({
+      alignItems: 'stretch',
+      backgroundColor: ColorPalette.white,
+      border: 'none',
+      boxShadow: isOpen && '1px 1px 3px rgba(29, 29, 29, 0.125)',
+      cursor: disabled ? 'not-allowed' : !isLoading && 'pointer',
+      display: 'flex',
+      flexDirection: 'column',
+      opacity: disabled && 0.85,
+      padding: 0,
+      width: '100%',
+    }),
+}
+
+interface ISearchableDropdownProps {
+  /** If true, than the field can be cleared */
+  clearable?: boolean
+  /** Set dropdown into disabled state */
+  disabled?: boolean
+  /** Icon on the left of the input field */
+  icon?: IconType
+  /** The dropdown items to show */
+  items: IDropdownItem[]
+  /** Initially selected item - this value is uncontrolled */
+  initialSelectedItem?: IDropdownItem
+  /** The loading state of the component, e.g fetching data. */
+  isLoading?: boolean
+  /** A floating label */
+  label?: string
+  /** The "Load more" text */
+  loadMoreText?: string
+  /** The height of the menu in pixels. By default: Search input and 3 items */
+  menuHeight?: number
+  /** For forms */
+  name?: string
+  /** The text is shown if no result was found */
+  noResultsText?: string
+  /** Callback triggered when clicking on "Load more" in items list */
+  onLoadMore?: () => void
+  /** Callback triggered when clearing the selection. */
+  onSelect: (item: IDropdownItem) => void
+  /** Callback triggered when search value changes */
+  onSearch?: (event: SyntheticEvent) => void
+  /** The placeholder displayed in the input field. */
+  placeholder?: string
+  /** If "top", then the list should be reversed and extended upwards, if "bottom" (default) then downwards */
+  placement?: EDropdownDirection
+  /** The search term */
+  searchTerm?: string
+  /** Selected item - this item can be controlled */
+  selectedItem?: IDropdownItem
+}
+
+const SearchableDropdown = ({
+  clearable = false,
+  disabled = false,
+  icon,
+  items,
+  initialSelectedItem,
+  isLoading = false,
+  label = '',
+  loadMoreText = '',
+  menuHeight = INPUT_FIELD_HEIGHT * 4,
+  name = '',
+  noResultsText,
+  onLoadMore = noop,
+  onSearch,
+  onSelect,
+  placeholder = '',
+  placement = EDropdownDirection.BOTTOM,
+  searchTerm = '',
+  selectedItem,
+}: ISearchableDropdownProps) => {
+  const searchRef: RefObject<HTMLInputElement> = createRef()
```

Comment:
Should rather be: 
```suggestion
  const searchRef = useRef<HTMLInputElement>()
```

Right?

Reply:
👍  Done